powermon/commands/reading_definition.py

```diff
@@ -326,110 +375,12 @@ def reading_from_raw_response(self, raw_value) -> list[Reading]:
-        return self.description
-
-class ReadingDefinitionOption(ReadingDefinition):
-    def __init__(self, index: int, description: str, options: list[str], device_class: str = None, state_class: str = None, icon: str = None):
-        self.index = index
-        self.description = description
-        self.options = options
-        self.device_class = device_class
-        self.state_class = state_class
-        self.icon = icon
-
-    def is_valid_response(self, value) -> bool:
-        return value in self.options
-
-    def translate_raw_response(self, raw_value) -> str:
-        value = int(raw_value.decode())
-        return self.options[value]
-
-    def reading_from_raw_response(self, raw_value) -> list[Reading]:
-        value = self.translate_raw_response(raw_value)
-        return [
-            Reading(
-                data_name=self.description,
-                data_value=value,
-                data_unit="",
-                device_class=self.device_class,
-                state_class=self.state_class,
-                icon=self.icon,
-            )
-        ]
-
-    def get_description(self) -> str:
-        return self.description
-
-class ReadingDefinitionBytes(ReadingDefinition):
-    # Seems to be a catch-all for responses that are not properly implemented
-    def __init__(self, index: int, description: str, unit: str, device_class: str = None, state_class: str = None, icon: str = None):
-        self.index = index
-        self.description = description
-        self.unit = unit
-        self.device_class = device_class
-        self.state_class = state_class
-        self.icon = icon
-
-    def is_valid_response(self, value) -> bool:
-        return True
-
-    def translate_raw_response(self, raw_value) -> str:
-        return raw_value.decode()
-
-    def reading_from_raw_response(self, raw_value) -> list[Reading]:
-        value = self.translate_raw_response(raw_value)
-        return [
-            Reading(
-                data_name=self.description,
-                data_value=value,
-                data_unit="",
-                device_class=self.device_class,
-                state_class=self.state_class,
-                icon=self.icon,
-            )
-        ]
-
-    def get_description(self) -> str:
-        return self.description
-
-class ReadingDefinitionFloat(ReadingDefinition):
-    def __init__(self, index: int, description: str, unit: str, device_class: str = None, state_class: str = None, icon: str = None):
-        self.index = index
-        self.description = description
-        self.unit = unit
-        self.device_class = device_class
-        self.state_class = state_class
-        self.icon = icon
+class ReadingDefinitionTemperature(ReadingDefinition):
+    def __init__(self, index: int, name: str, description: str,  response_type: ResponseType, device_class: str = None, state_class: str = None, icon: str = None):
+        #TODO: find a way to make the unit configurable
```

Comment:
Also need to consider how to 'normalize' the input - ie remember the response that was in kelvin (and needing /10 to get standard K)

Reply:
My current thinking is that it will be ResponseType.INT_DECIKELVIN (maybe just DECIKELVIN) or something like that then the ReadingType.TEMPERATURE will handle the conversion logic.

Reply:
Agree. I think the first option so we could have other types eg uint or hex or bytes or whatever if needed